src/EVM/Traversals.hs

```diff
@@ -211,7 +211,7 @@ foldExpr f acc expr = acc <> (go expr)
 mapProp :: (forall a . Expr a -> Expr a) -> Prop -> Prop
 mapProp f = \case
   PBool b -> PBool b
-  PEq a b -> PEq (mapExpr f (f a)) (mapExpr f (f b))
+  PEq a b -> mkPEq (mapExpr f (f a)) (mapExpr f (f b))
```

Comment:
This is just a traversal, we should not change things here.

Reply:
But then this could break the invariant that were are trying to establish, no?
Because it can transform the second argument to `Lit` and then we would again end up with `PEq` with constant on RHS instead of LHS.

Reply:
Oh, actually, right. I find it a bit strange that it'd change the thing on its own, but at the same time it'd be painful to have to deal with it every time otherwise... So yeah, you are right. Sorry!

Reply:
I guess this is a design question. I could imagine different approaches. Just thought that we would really like to establish this invariant :)